src/decorators.ts

```diff
@@ -1,15 +1,15 @@
-import * as express from "express";
 import { inject, injectable, decorate } from "inversify";
 import { interfaces } from "./interfaces";
 import { TYPE, METADATA_KEY, PARAMETER_TYPE } from "./constants";
+import { getOrCreateMetadata } from "./utils";
 
 export const injectHttpContext = inject(TYPE.HttpContext);
 
-export function controller(path: string, ...middleware: interfaces.Middleware[]) {
+export function controller(path: string, ...middlewares: interfaces.Middleware[]) {
```

Comment:
`middleware` is plural, you don't need to add the `s` at the end

Reply:
ok, initially I had to do that because of `no-shadow` tslint rule

Reply:
Aahh then it's legit I guess :)